### PR TITLE
hardinfo 0.5.1 rework 

### PR DESCRIPTION
**Pull request by CamberLoid**
Topic Description
-----------------

* Latest commit on upstream source tree has made multiple changes comparing to our version, including supports for various architectures, bugfixes, and adaption to modern envrionments. The package will now use snapshot of latest commit from master branch from Github instead of debian's source code.
* The package should use ABTYPE=CMAKENINJA from this commit.

Package(s) Affected
-------------------

- `hardinfo`: Update
- `sysbench`: new
- `concurrencykit`: new

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Review by Nyovelt — COMMENTED** [review]
Built and worked success on my amd64 devices

**Review by Nyovelt — DISMISSED** [review]

File: extra-libs/concurrencykit/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,9 @@
+PKGNAME=concurrencykit
+PKGDES="Concurrency Kit C Library, with related algorithms and data structures"
```
Comment (MingcongBai) [inline extra-libs/concurrencykit/autobuild/defines]: C library to aid concurrent application development

File: extra-utils/sysbench/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,14 @@
+PKGNAME=sysbench
+PKGSEC=utils
+PKGDES="Scriptable database and system performance benchmark, based on LuaJIT"
```
Comment (MingcongBai) [inline extra-utils/sysbench/autobuild/defines]: Scriptable database and system performance benchmark

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by hwspeedy** [comment]
Hi AOSC,

We have released new version of hardinfo as community edition hardinfo2.

Please revert to debian source code for hardinfo2 or make an updated package from community edition.

github source: https://github.com/hardinfo2/hardinfo2
webpage: https://hardinfo2.org

**Pull request merged**